lib/web/fetch/index.js

```diff
@@ -133,116 +178,147 @@ function fetch (input, init = undefined) {
   let requestObject
 
   try {
-    requestObject = new Request(input, init)
-  } catch (e) {
-    p.reject(e)
-    return p.promise
-  }
+    publishIfContext('start', context)
 
-  // 3. Let request be requestObject’s request.
-  const request = requestObject[kState]
+    try {
+      requestObject = new Request(input, init)
+    } catch (e) {
+      p.reject(e)
+      publishIfContext('fetchError', context, e)
```

Comment:
You can omit this if you publish to the `fetchError` channel in `handlePromise` instead.

```suggestion
```